clients/imodels-client-management/src/base/types/HttpRequestRetryPolicy.ts

```diff
@@ -0,0 +1,35 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+export interface OnFailedRequestParams {
+  /** The number of already invoked retries, starting from 0. */
+  retriesInvoked: number;
+  /** The error that was thrown when sending the HTTP request. */
+  error: unknown;
```

Comment:
this is very vague, implementation of this interface would be very problematic if we do not guarantee any information that we return with the error. let's say user uses our internal client, but wants to override retry policy. user does not know what kind of errors our axios client throws. we could define an interface that closely matches the error we get from `axios`, so that at least the failure would have an optional `response` with `code` and `body`.

Reply:
But we don't know about the thrown error. Since we catch `unknown` error, we can only do a check if it's an axios error. Clients can just use `isIModelsApiError` and get the information that they need (as it is done in `DefaultHttpRequestRetryPolicy`).

Reply:
> Clients can just use isIModelsApiError and get the information that they need

good point which brings out an issue with having retries in `OperationsBase` level - we currently catch error in axios client, adapt it and throw adapted error. If it is some network issue error, we will loose that information here - https://github.com/iTwin/imodels-clients/blob/0594ffac53e05d8d0dda1baef56ff77e86f8eaba/clients/imodels-client-management/src/base/internal/IModelsErrorParser.ts#L68
`IModelsError` is supposed to be loggable and readable by client users, but it removes quite a lot of information returned by the original error thrown by rest library which is crucial for retry logic. I think we should still accept `HttpRequestRetryPolicy` in the constructor of `IModelsClient` but it should be `AxiosRestClient` that uses it, not `OperationsBase`

Reply:
updated as talked